File: MoveMiddleToHead/MoveMiddleToHead.py
from Solution import *
from crio.ds.List.ListNode import *


def print_list(head):
    while head != None:
        print(head.val, end = " ")
        head = head.next

def main():
   n = int(input().strip())
   lst = None
   if n > 0:
       lst = input().strip().split()
       lst = [int(x) for x in lst]
       lst = createList(lst)
   
   print_list(move_middle_to_head(lst))

if __name__ == '__main__':
    main()

File: MoveMiddleToHead/MoveMiddleToHead.java
import java.util.*;
import crio.ds.List.ListNode;



class MoveMiddleToHead{
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int n=sc.nextInt();
		ArrayList<Integer> input=new ArrayList<Integer>();
		for(int i=0;i<n;i++)
			input.add(sc.nextInt());
		ListNode head=ListNode.createList(input);
		Solution so=new Solution();
		ListNode start=so.moveMiddleToHead(head);
		List<Integer> ans=ListNode.extractList(start);
		for(int i=0;i<ans.size();i++)
		{
			System.out.print(ans.get(i)+" ");
		} 

	}
}
